src/main/java/org/jvnet/hudson/test/MemoryAssert.java

```diff
@@ -146,13 +148,16 @@ public static void assertGC(WeakReference<?> reference) {
     }
 
     /**
+     * <strong>Assumes Java runtime is &le; Java 8. I.e. tests will be skipped if Java 9+</strong>
      * Forces GC by causing an OOM and then verifies the given {@link WeakReference} has been garbage collected.
      * @param reference object used to verify garbage collection.
      * @param allowSoft if true, pass even if {@link SoftReference}s apparently needed to be cleared by forcing an {@link OutOfMemoryError};
      *                  if false, fail in such a case (though the failure will be slow)
      */
     @SuppressWarnings("DLS_DEAD_LOCAL_STORE_OF_NULL")
     public static void assertGC(WeakReference<?> reference, boolean allowSoft) {
+        // Disabled on Java 9+, because below will call Netbeans Insane Engine, which in turns tries to call setAccessible
+        assumeTrue(System.getProperty("java.specification.version").startsWith("1."));
```

Comment:
I'm not sure if lib-version-number is in scope here, but if so, it has `io.jenkins.lib.versionnumber.JavaSpecificationVersion` which could be used to make this `JavaSpecificationVersion.forCurrentJVM().isOlderThanOrEqualTo(JavaSpecificationVersion.JAVA_8);` as @alecharp did in jenkinsci/script-security-plugin#238.

Reply:
Done, good idea